### PR TITLE
Feat/components dropdown button

### PR DESCRIPTION
**Pull request by connoratrug**

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/ButtonDropdown.vue [hunk 1]
```diff
@@ -1,65 +1,123 @@
...
+    icon: {
+      type: String,
+      required: false,
+      default: "",
+    },
     isMenuItem: Boolean,
+    placement: {
+      type: String,
+      required: false,
+      default: () => "bottom-start",
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/ButtonDropdown.vue:43]: Why is this a function, while type says string?

**Review by chinook25 — APPROVED** [review]

File: apps/molgenis-components/src/components/forms/ButtonDropdown.vue [hunk 2]
```diff
@@ -1,65 +1,123 @@
...
+      popperInstance: null,
     };
   },
   methods: {
-    toggle() {
+    async toggle() {
       this.display = !this.display;
+      if (this.display) {
+        await this.$nextTick();
+        const dropDownBtn = this.$refs["showInputButton"];
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/ButtonDropdown.vue:57]: looks rather error prone to use refs. Now there is a lot a dev needs to know while making changes? Is there not a better way?

File: apps/molgenis-components/package.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
   "dependencies": {
     "@vue/compiler-sfc": "3.2.37",
     "axios": "0.27.2",
+    "popper.js": "1.16.1",
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/package.json:25]: > do we really need another dependency?

Yes this is tricky stuff cross browser, if you have an alternative ? this is used by bootstrap to handle popovers

**Comment by connoratrug** [comment]
> It might be nice to add a bit of a border to the popover. It's a bit hard to see over the text in the example.

the  popover has a border
Screenshot 2022-07-04 at 11 27 04

**Pull request merged**